code.gs

```diff
@@ -430,148 +469,190 @@ function checkEvents (testDate) {
+        } else {
+          line.push(' - ', _('Years'), ': ');
         }
-        label = phoneField.getLabel();
-        if (label !== '') {
-          line.push('[', beautifyLabel(label), '] ');
+        line.push(self.age[dateLabel]);
+      }
+
+      if (self.email !== '' || typeof self.phoneFields !== 'undefined') {
+        line.push(' (');
+        // Email address.
+        if (self.email !== '') {
+          line.push(self.email);
         }
-        line.push(phoneField.getPhoneNumber());
-      });
-      line.push(')');
-    }
-    line.push('\n');
-    return line;
+        // Phone numbers.
+        self.phoneFields.forEach(function (phoneField, i) {
+          var label;
+          if (i !== 0 || self.email !== '') {
+            line.push(' - ');
+          }
+          label = phoneField.getLabel();
+          if (label !== '') {
+            line.push('[', beautifyLabel(label), '] ');
+          }
+          line.push(phoneField.getPhoneNumber());
+        });
+        line.push(')');
+      }
+      lines.push(line);
+    });
+    return lines.map(function (x) { return x.join(''); }).join('\n');
   };
 
   /*
    * Use the extracted data to build a line of HTML text displaying all the
    * collected data about the contact.
    */
-  this.getHtmlLine = function () {
-    var line, imgCount;
-
-    line = [];
-
-    line.push('<li>');
-    // Profile photo.
-    if (this.photo !== '') {
-      imgCount = Object.keys(inlineImages).length;
-      inlineImages['contact-img-' + imgCount] = UrlFetchApp.fetch(this.photo).getBlob().setName('contact-img-' + imgCount);
-      line.push('<img src="cid:contact-img-' + imgCount + '" style="height:1.4em;margin-right:0.4em" />');
-    }
-    // Full name.
-    line.push(this.fullName);
-    // Nickname.
-    if (this.nickname !== '') {
-      line.push(' &quot;', this.nickname, '&quot;');
-    }
-    // Age.
-    if (this.age !== '') {
-      line.push(' - ', _('Age'), ': ', this.age);
-    }
-    if (this.email !== '' || typeof this.phoneFields !== 'undefined') {
-      line.push(' (');
-      // Email address.
-      if (this.email !== '') {
-        line.push(this.email);
+  self.getHtmlLine = function () {
+    var lines;
+    lines = [];
+    self.dateLabels.forEach(function (dateLabel) {
+      var line, imgCount;
+      line = ['<li>'];
+      // Profile photo.
+      if (self.photo !== '') {
+        imgCount = Object.keys(inlineImages).length;
+        inlineImages['contact-img-' + imgCount] = UrlFetchApp.fetch(self.photo).getBlob().setName('contact-img-' + imgCount);
+        line.push('<img src="cid:contact-img-' + imgCount + '" style="height:1.4em;margin-right:0.4em" />');
       }
-      // Phone fields.
-      this.phoneFields.forEach(function (phoneField, i) {
-        var label;
-
-        if (i !== 0 || this.email !== '') {
-          line.push(' - ');
+      // Custom label
+      if (self.eventType === 'CUSTOM') {
+        line.push('<' + dateLabel + '> ');
```

Comment:
`<` and `>` need to be escaped in the HTML, otherwise the result is going to be interpreted as a tag and won't be rendered in the message.